app/models/archive_document/base.rb

```diff
@@ -7,16 +7,31 @@ def initialize(document)
     @document = document
   end
 
-  def add_to_archive(zipfile)
+
+  def add_to_archive_and_csv(zipfile, csv)
```

Comment:
This can still be called `add_to_archive` since the CSV file is *in* the archive :stuck_out_tongue: